client/extensions/woocommerce/woocommerce-services/components/text-field/index.js

```diff
@@ -24,19 +24,22 @@ const TextField = ( {
 	updateValue,
 	error,
 	className,
+	required,
+	defaultValue
 } ) => {
-	const handleChangeEvent = event => updateValue( event.target.value );
+	const handleChangeEvent = useCallback(event => updateValue( event.target.value, event ), [updateValue]);
```

Comment:
- Wrapped the `onChange` with a `useCallback` to reduce unwanted re-renders.
- Passing the original `event` as the second argument to the `updateValue` prop, so it can be leveraged by consumers